src/SIL.Harmony/SnapshotWorker.cs

```diff
@@ -151,6 +154,7 @@ private async ValueTask MarkDeleted(Guid deletedEntityId, Commit commit)
             .Select(s => s.EntityId)
             .ToArrayAsync());
         //snapshots from the db might be out of date, we want to use the most up to date data in the worker as well
+        toRemoveRefFromIds.UnionWith(_rootSnapshots.Values.Where(predicate).Select(s => s.EntityId));
```

Comment:
I'm torn on this change. I was initially calling dbContext.SaveChanges() everytime I applied a snapshot to the projected tables and then I needed this change for some of my new tests. After moving the SaveChanges() outside the loop, I stopped needing this.
I don't have time today to dig into exactly why that was the case.

Reply:
I'm going to remove it as this may rewrite history. Here's a scenario all in a single commit
1. create schoolbus with component references
2. delete bus, creates 2 pending snapshots, one for bus and one to remove the bus component from schoolbus
3. now on this line our original schoolbus gets the bus reference removed. That's applying a change to a snapshot before it happened.

Thought... there's kinda a problem as there should probably be only 1 snapshot per commit per object. So maybe this isn't a problem?

Reply:
wrote a test and proved that this was needed